src/AdoPat/PatClient.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// An abstraction over common operations with the Azure DevOps PAT Lifecycle Management REST API.
+    /// </summary>
+    public class PatClient
+    {
+        private const int PageSize = 100; // Using the maximum allowable page size allows us to reduce HTTP calls.
+        private const bool AllOrgs = false; // Azure DevOps have recommended we always set this to false.
+
+        private ITokensHttpClientWrapper client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatClient"/> class.
+        /// </summary>
+        /// <param name="client">Any class which implements the <see cref="ITokensHttpClientWrapper"/> interface.</param>
+        public PatClient(ITokensHttpClientWrapper client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Creates a new PAT.
+        /// </summary>
+        /// <param name="patTokenCreateRequest">
+        /// XXX: Not documented by Visual Studio client libraries. Should we depend on this or
+        /// should this method use the corresponding primitives as arguments? Not yet clear.
+        /// </param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatTokenResult"/>.</returns>
+        public async Task<PatToken> CreatePatAsync(
+            PatTokenCreateRequest patTokenCreateRequest,
+            CancellationToken cancellationToken = default)
+        {
+            var patTokenResult = await this.client.CreatePatAsync(
+                patTokenCreateRequest,
+                cancellationToken: cancellationToken)

```

Comment:
Curious about what's the default timeout for all these requests how are we handling timeout/network exceptions

Reply:
That is an _excellent_ question and the honest answer is that it's pretty obscured by the `Microsoft.VisualStudio.Services.Client` library, so I'll have to do some spelunking before I can give a proper answer... 😞

Reply:
Is it possible to validate it through tests?

Reply:
Not without actually making the API calls. And even then it would be difficult, because if there are retries in the client library they're likely to not be exposed to us and just have their own internal retry loops instead.